packages/element/src/lib/element.ts

```diff
@@ -1,55 +1,60 @@
 import { metadataStore } from './metadata.js';
 
-export function element<Target extends CustomElementConstructor>(
-  Base: Target,
-  ctx: ClassDecoratorContext<Target>
-) {
-  const meta = metadataStore.read(ctx.metadata);
-
-  ctx.addInitializer(function (this: Target) {
-    if (meta.tagName) {
-      const val = meta.tagName(this);
-
-      if (!customElements.get(val)) {
-        customElements.define(val, this);
+export interface ElementOpts {
+  tagName?: string;
+}
+
+export function element(opts?: ElementOpts) {
+  return <Target extends CustomElementConstructor>(
+    Base: Target,
+    ctx: ClassDecoratorContext<Target>
+  ) => {
+    const meta = metadataStore.read(ctx.metadata);
+
+    ctx.addInitializer(function (this: Target) {
+      if (opts?.tagName) {
+        if (!customElements.get(opts.tagName)) {
+          customElements.define(opts.tagName, this);
+        }
       }
-    }
-  });
+    });
+
+    return class JoistElement extends Base {
```

Comment:
Again would be nice to preserve the orignal classname for debuggability.

Reply:
I will have to see if there is an easy way. decorators work the opposite direction from mixins. I could remove JoistElement but that would just be an anonymous class.